xrviz/fields.py

```diff
@@ -60,20 +58,21 @@ def set_data(self, data):
 
     def setup(self, var):
         self.agg_selectors.clear()  # To empty previouly selected value from selector
-        self.agg_graph[0] = pn.Spacer(name='Agg Graph')  # To clear Agg Graph upon selection of new variable
 
         if self.is_dataset:
             if isinstance(var, str):
                 self.var = var
             else:
                 self.var = var[0]
             self.var_dims = list(self.data[var].dims)
+            self.indexed_coords = set(self.var_dims) & set(self.data[var].coords)
```

Comment:
This appears to be repeated code.

Same comments above about using set methods rather than operators.
Also, is it weird that the first one remains a set , but the other two are lists?

Reply:
I have replaced operators with set methods. Since `pn.widget.Select` accepts only `list` or `dict` as options, `self.sel_options` has to be a list and not set.

Reply:
The result of the `list(set(...))` has arbitrary ordering, you might want `sorted(set(...))` instead.

Reply:
Sorted has been applied to end result, i.e. just before passing as options to x and y.